### PR TITLE
FIXES #779 | Open unhandled twitter deeplinks in an external webview

### LINKED ISSUE #779: Open unhandled twitter deeplinks in an external webview
Currently, if a user opens a Twitter url with harpy but harpy does not have a matching path or can't map the url (e.g. a link to a DM) - harpy will just navigate to the home page.

Instead the url should just be opened in an external webview.

#### Requirements & technical information

- In `handle_redirect.dart` we navigate to home if a requested location doesn't exist (line 21).
- Instead use the `launcherProvider` from `url_launcher.dart` to open the path as twitter url in an external webview

### PR DESCRIPTION
**Pull request by Dhi13man**
Created url_launcher based redirect for unparsable twitter URLs.

This is the fix for Issue [#779](https://github.com/robertodoering/harpy/issues/779).

### CODE REVIEW

File: lib/core/router/handle_redirect.dart [hunk 1]
```diff
@@ -68,7 +71,11 @@ String? handleRedirect(Ref ref, GoRouterState state) {
     final mappedLocation = _mapTwitterPath(state);
     if (mappedLocation != null) return mappedLocation;
 
-    // if the location doesn't exist navigate to home instead
+    // if the location doesn't exist, launch it and navigate to home instead
+    final launcher = ref.watch(launcherProvider);
+    launcher(state.location).onError((error, stackTrace) {
+      log('failed to launch url: $error');
+    });
```
Comment (robertodoering) [inline lib/core/router/handle_redirect.dart:78]: - `state.location` only includes the path, we need to prefix the twitter url
- the error is handled by the launcherProvider, so no need for the onError handler

**Review by robertodoering — CHANGES_REQUESTED** [review]

File: lib/core/router/handle_redirect.dart (hunk 1 above)
Comment (Dhi13man) [reply lib/core/router/handle_redirect.dart]: Updated. Please check and advice.

File: lib/core/router/handle_redirect.dart [hunk 2]
```diff
@@ -31,6 +32,9 @@ adb shell am start -a android.intent.action.VIEW \
  -d "https://twitter.com/harpy_app/followers"
 */
 
+/// Twitter prefix URL constant
+const String twitterPrefix = 'https://twitter.com/';
+
```
Comment (robertodoering) [inline lib/core/router/handle_redirect.dart:37]:
```suggestion
```

File: lib/core/router/handle_redirect.dart [hunk 3]
```diff
@@ -2,6 +2,7 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:go_router/go_router.dart';
 import 'package:harpy/components/components.dart';
 import 'package:harpy/core/core.dart';
+import 'package:harpy/rby/core/url_launcher.dart';
```
Comment (robertodoering) [inline lib/core/router/handle_redirect.dart:5]:
```suggestion
```
Please rebase, it will make this import obsolete

**Comment by Dhi13man** [comment]
updated

**Comment by Dhi13man** [comment]
Removed import

**Comment by robertodoering** [comment]
Looks good now, ty

**Review by robertodoering — APPROVED** [review]

**Pull request merged**